libs/community/langchain_community/vectorstores/yellowbrick.py

```diff
@@ -46,67 +56,144 @@ def __init__(
         self._table = table
         self._embedding = embedding
-        self._connection = psycopg2.connect(connection_string)
+        self._max_embedding_len = None
+        self.idle_threshold_seconds = idle_threshold_seconds
+        self._connection = psycopg2.connect(self.connection_string)
+        self.last_used_time = time.time()
+        self._check_database_utf8()
 
-        self.__post_init__()
+        if drop:
+            self._drop_table(self._table)
+            self._drop_table(self._table + self.CONTENT_TABLE)
+            self._drop_lsh_index_tables()
 
-    def __post_init__(
-        self,
-    ) -> None:
-        """Initialize the store."""
-        self.check_database_utf8()
-        self.create_table_if_not_exists()
+        self._create_table()
+
+        self._use_lsh = use_lsh
+        self._hamming_distance = 0
+
+        if seed is not None:
+            random.seed(seed)
+        self._seed = seed
 
     def __del__(self) -> None:
         if self._connection:
             self._connection.close()
 
-    def create_table_if_not_exists(self) -> None:
+    @property
+    def use_lsh(self) -> bool:
+        return self._use_lsh
+
+    @use_lsh.setter
+    def use_lsh(self, use_lsh: bool) -> None:
+        self._use_lsh = use_lsh
+        if self._use_lsh:
+            self._create_lsh_index_tables()
+
+    @property
+    def hamming_distance(self) -> bool:
+        return self._hamming_distance
+
+    @hamming_distance.setter
+    def hamming_distance(self, hamming_distance: int) -> None:
+        self._hamming_distance = hamming_distance
+
+    def _get_connection(self):
+        import psycopg2
+        from psycopg2 import Error, OperationalError
+
+        current_time = time.time()
+        if self._connection.closed:
+            self._connection = psycopg2.connect(self.connection_string)
+            self.last_used_time = current_time
+        elif (current_time - self.last_used_time) > self.idle_threshold_seconds:
+            try:
+                with self._connection.cursor() as cursor:
+                    cursor.execute("SELECT 1")
+            except (OperationalError, Error):
+                self._connection = psycopg2.connect(self.connection_string)
+            self.last_used_time = current_time
+
+        return self._connection
+
+    def _create_table(self) -> None:
         """
         Helper function: create table if not exists
         """
         from psycopg2 import sql
 
-        cursor = self._connection.cursor()
+        cursor = self._get_connection().cursor()
         cursor.execute(
             sql.SQL(
-                "CREATE TABLE IF NOT EXISTS {} ( \
-            id UUID, \
-            embedding_id INTEGER, \
-            text VARCHAR(60000), \
-            metadata VARCHAR(1024), \
-            embedding FLOAT)"
-            ).format(sql.Identifier(self._table))
+                """
+                CREATE TABLE IF NOT EXISTS {t} (
+                doc_id UUID NOT NULL,
```

Comment:
this is a breaking change in the schema? Is there a migration for users to use?

Reply:
Good point. I've introduced a new function to migrate from the prior schema to the updated schema.